src/Components/Components/src/Forms/InputComponents/InputDate.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// An input component for editing date values.
+    /// Supported types are <see cref="DateTime"/> and <see cref="DateTimeOffset"/>.
+    /// </summary>
+    public class InputDate<T> : InputBase<T>
+    {
+        const string dateFormat = "yyyy-MM-dd"; // Compatible with HTML date inputs
+
+        [Parameter] string ParsingErrorMessage { get; set; } = "The {0} field must be a date.";
```

Comment:
I would like to have a chance to localize messages like this and don't pass them as parameters each time.

Reply:
Fair request. Localisation is tracked as #5471, so I've added a note there about this.

In the meantime, if you want to change your `ParsingErrorMessage` values centrally, you can subclass `InputDate` and set a different `ParsingErrorMessage` in `OnInit`.